### PR TITLE
Adding Disabled Props for iOS

### PR DESCRIPTION
**Pull request by gedeagas**
# Summary
Disabled prop is only supported in android. Basic functionality like disabled should be available for both platforms.
This pr adds support for disabled props on iOS. The implementation is done via **JS-level**. 

Related issue. https://github.com/react-native-community/react-native-checkbox/issues/56

## Test Plan
Adding jest snapshot testing for iOS Checkbox with disabled props. It should render the checkbox with `pointerEvents="none"`

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |
| Android |    ✅    |

## Checklist
- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: src/js/__test__/CheckBox.test.tsx [hunk 1]
```diff
@@ -38,6 +38,14 @@ describe('render IOS <Checkbox />', () => {
     expect(tree).toMatchSnapshot();
   });
 
+  it('renders disabled IOS Checkbox', () => {
```
Comment (nicholaslee119) [inline src/js/__test__/CheckBox.test.tsx:41]: please add tests for `disable={false}` and `none disable`

Comment (gedeagas) [reply src/js/__test__/CheckBox.test.tsx:41]: will do. thx for the review

Comment (gedeagas) [reply src/js/__test__/CheckBox.test.tsx:41]: updated @nicholaslee119, kindly check again.
